libsplinter/src/node_id/store/error.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::From;
+use std::error::Error;
+use std::fmt::Display;
+
+#[derive(Debug)]
+pub enum NodeIdStoreError {
+    DieselError(diesel::result::Error),
```

Comment:
This should use our top level error types instead of the diesel error directly. See https://github.com/Cargill/splinter/blob/main/libsplinter/src/admin/store/error.rs for an example.

Reply:
It uses now uses InternalError